src/main/java/kayu/Kayu.java

```diff
@@ -0,0 +1,108 @@
+package kayu;
+
+import java.util.List;
+
+import kayu.commands.Command;
+import kayu.exception.DukeException;
+import kayu.exception.StorageException;
+import kayu.parser.Parser;
+import kayu.service.Logger;
+import kayu.service.TaskList;
+import kayu.storage.Storage;
+import kayu.task.Task;
+
+/**
+ * Holds the main logic for Kayu project (CS2103T's iP).
+ * Kayu is a task manager for users that love the command line interface.
+ * Code structure inspiration from <code>se-edu</code>'s addressbook-level2 code on GitHub
+ * <a href="https://github.com/se-edu/addressbook-level2">addressbook-level2</a>.
+ * @author muhammad-khair, damithc, j-lum
+ */
+public class Kayu {
+
+    private static final String GREETING = "Hello!\n"
+            + "I'm Kayu, your alternative personal task management to Duke!\n"
+            + "What can I do for you?";
+    
+    private final Parser parser = new Parser();
+    private final TaskList taskList = new TaskList();
+    private final Storage storage = new Storage();
+    private final Logger logger = new Logger();
+    
+    private boolean isRecentCommandBye = false;
+
+    /**
+     * Returns whether the recent command parsed and executed is a {@link kayu.commands.ByeCommand}.
+     *
+     * @return Boolean true if recent command is a {@link kayu.commands.ByeCommand}, else false.
+     */
+    public boolean isRecentCommandBye() {
+        return isRecentCommandBye;
+    }
```

Comment:
Maybe instead of making the isRecentCommandBye private, you can make the variable package protected and just call on it for other functions